crates/ruff/src/rules/isort/categorize.rs

```diff
@@ -48,9 +48,20 @@ pub fn categorize(
     extra_standard_library: &BTreeSet<String>,
     target_version: PythonVersion,
 ) -> ImportType {
+    let module_base = module_name.split('.').next().unwrap();
     let (import_type, reason) = {
         if level.map_or(false, |level| *level > 0) {
             (ImportType::LocalFolder, Reason::NonZeroLevel)
+        } else if module_base == "__future__" {
+            (ImportType::Future, Reason::Future)
+        } else if known_first_party.contains(module_name) {
+            (ImportType::FirstParty, Reason::KnownFirstParty)
+        } else if known_third_party.contains(module_name) {
+            (ImportType::ThirdParty, Reason::KnownThirdParty)
+        } else if known_local_folder.contains(module_name) {
+            (ImportType::LocalFolder, Reason::KnownLocalFolder)
+        } else if extra_standard_library.contains(module_name) {
+            (ImportType::StandardLibrary, Reason::ExtraStandardLibrary)
```

Comment:
What does isort do if you have:

```
known-first-party = ["foo"]
known-third-party = ["foo.bar"]
```

And then you import `foo.bar.baz`. Does that get classified as first- or third-party?

Reply:
```ini
[settings]
known_first_party=foo
known_third_party=foo.bar
```
results in:
```python
import sys

import foo.bar
import foo.bar.baz
from foo.bar import blah, blub
from foo.bar.baz import something

import foo
from foo import bar, baz
```

So my MR currently has this diff:
```diff
ruff --select I --fix --diff foo.py 
--- foo.py
+++ foo.py
@@ -1,9 +1,9 @@
 import sys
 
 import foo.bar
-import foo.bar.baz
 from foo.bar import blah, blub
-from foo.bar.baz import something
 
 import foo
+import foo.bar.baz
 from foo import bar, baz
+from foo.bar.baz import something

Would fix 1 error.
```